src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -315,6 +401,51 @@ public bool IsSolutionOpen
             return Path.IsPathRooted(globalPackagesFolder);
         }
 
+        public async Task<bool> DoesNuGetSupportsAnyProjectAsync()
+        {
+            // Do NOT initialize VSSolutionManager through this API (by calling EnsureInitializeAsync)
+            // This is a fast check implemented specifically for right click context menu to be
+            // quick and does not involve initializing VSSolutionManager. Otherwise it will create
+            // hang issues for right click on solution.
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            // first check with DTE, and if we find any supported project, then return immediately.
+            var dte = _serviceProvider.GetDTE();
+
+            var isSupported =  EnvDTESolutionUtility.GetAllEnvDTEProjects(dte)
+                .Where(EnvDTEProjectUtility.IsSupported)
+                .Any();
+
+            if (isSupported)
+            {
+                return true;
+            }
+
+            var deferredProjects = await GetDeferredProjectsAsync();
+
+            foreach (var project in deferredProjects)
+            {
+                try
+                {
+                    var vsProjectAdapter = await _vsProjectAdapterProvider.CreateAdapterForDeferredProjectAsync(project);
+
+                    if (vsProjectAdapter.IsSupported)
+                    {
+                        // as soon as we find a supported project, we returns without checking for all the projects.
+                        return true;
+                    }
+                }
+                catch (Exception e)
+                {
+                    // Ignore failed projects.
+                    _logger.LogWarning($"The project {project} failed to initialize as a NuGet project.");
```

Comment:
It's not a customer facing warning, rather being written in Activity logs so doesn't need to be localized. We're already logging the same error message for the original API with full initialization.